docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,220 @@
+
+You could use `substreams-sink-files` to sink data in `JSONL` format to a [Google Cloud Storage (GCS)](https://cloud.google.com/storage) bucket and configure a BigQuery Transfer job to run every 15 minutes. The scheduled job ingests the new files found in the GCS bucket where the data, extracted by the Substreams, was written.
+
+### Accompanying code example
+
+The accompanying Substreams module associated with this documentation is responsible for extracting a handful of data fields from the Block object injected into the Rust-based map module. The sink tool processes the extracted blockchain data line-by-line and outputs the data to the files-based persistence mechanism you've chosen.
+
+The accompanying code example extracts four data points from the Block object and packs them into the `substreams.sink.files.v1` protobuf's data model. The data is passed to the protobuf as a single line of plain text.
+
+Binary formats such as [Avro](https://avro.apache.org/) or [Parquet](https://parquet.apache.org/) is possible, however, support is not available. Contributions are welcome to help with support of binary data formats. [Contact the StreamingFast team on Discord](https://discord.gg/mYPcRAzeVN) to learn more and discuss specifics.
+
+## Download code and tools
+
+### Install `substreams-sink-files`
+
+Clone the `substreams-sink-files` repository to obtain the files required to work with the sink.
+
+```bash
+git clone https://github.com/streamingfast/substreams-sink-files.git substreams-sink-files-tutorial
+```
+
+To install the `substreams-sink-files` tool you need to run the command:
+
+```bash
+go install ./cmd/substreams-sink-files
+```
+
+Checking the version of `substreams-sink-files` will produce a message similar to:
+
+```bash
+substreams-sink-files version v0.2.0
+```
+
+Add the following lines to the computer's `~/.bashrc` configuration file and then restart the shell session to use the `substreams-sink-tool` from anywhere on your system.
+
+```bash
+export GOPATH=$HOME/go
+```
+
+### Accompanying code example
+
+The accompanying code example for this tutorial is available in the `substreams-sink-tool` respository. The Substreams project for the tutorial is located in the `docs/tutorial/` directory.
+
+**2. QUESTION**:
+<i>'Explanation of what is required from Substreams module perspective. We need to talk here about the Protobuf gen required to pull sf.substreams.sink.files.v1 model and modifications required, e.g. outputting a JSON model for each entity, one entity per line.'
+
+This is explained below in more detail. Do you want to move it here? Shouldn't we have them run the example first to make sure it's working properly on their machine. Then start explaining the code in more detail (as seen in the Substreams modifications section below)?</i>
```

Comment:
I'm not expecting people to follow the tutorial using the pre-made code, the pre-made code serves only as a reference for people. I'm expecting that by reading the instructions, they will be able to apply the knowledge and example directly to their Substreams.

Let's keep the flow like you did for now, explanation in next section will be good for now.

Reply:
Ok.